eng/common/scripts/Verify-RestApiSpecLocation.ps1

```diff
@@ -18,19 +18,33 @@ param (
 Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
 
 # This function is used to verify the 'require' and 'input-file' settings in autorest.md point to the main branch of Azure/azure-rest-api-specs repository
+# input-file may be: 
+# https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/purview/data-plane/Azure.Analytics.Purview.MetadataPolicies/preview/2021-07-01-preview/purviewMetadataPolicy.json
+# or 
+# https://github.com/Azure/azure-rest-api-specs/blob/0ebd4949e8e1cd9537ca5a07384c7661162cc7a6/specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2019-11-01-preview/account.json
 function Verify-AutorestMd([string]$fileUrl) {
-  if($fileUrl -match "^https://github.com/(?<repo>[^/]*/azure-rest-api-specs)/blob/(?<commit>[0-9a-f]{40})/(?<path>.*)") {
+  if($fileUrl -match "^https://raw.githubusercontent.com/(?<repo>[^/]*/azure-rest-api-specs)/(?<commit>[^\/]+(\/[^\/]+)*|[0-9a-f]{40})/(?<path>specification/.*)") {
```

Comment:
You should be able to make these into one regex instead of needing to split this into multiple conditions. 
```
^https://(raw.githubusercontent.com|github.com)/(?<repo>[^/]*/azure-rest-api-specs)/(blob|tree)?/(?<commit>[^\/]+(\/[^\/]+)*|[0-9a-f]{40})/(?<path>specification/.*)
```

Reply:
I thought about the combination of these two expressions but seem cannot be simplified. Anyway, I combined them with one simplified version with missing an edge case coverage. The below URL starting with `raw.githubusercontent.com` should not have `[blob|tree]` part after repository name.
```
https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/purview/data-plane/Azure.Analytics.Purview.MetadataPolicies/preview/2021-07-01-preview/purviewMetadataPolicy.json
```

Reply:
That is why I made `((blob|tree)/)?` optional in my regex example.

Reply:
I know, making it as optional cannot avoid this invalid case which should not have `blob` before `main`. However, it's ok to skip this case as this script is not the place to ensure the URL is valid.
```
https://raw.githubusercontent.com/Azure/azure-rest-api-specs/blob/main/specification/purview/data-plane/Azure.Analytics.Purview.MetadataPolicies/preview/2021-07-01-preview/purviewMetadataPolicy.json
```